src/SymfonyCache/KernelClient.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\SymfonyCache;
+
+use Http\Client\HttpAsyncClient;
+use Http\Discovery\MessageFactoryDiscovery;
+use Http\Message\ResponseFactory;
+use Http\Promise\FulfilledPromise;
+use Psr\Http\Message\RequestInterface;
+use Symfony\Bridge\PsrHttpMessage\Factory\HttpFoundationFactory;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Zend\Diactoros\ServerRequest;
+
+/**
+ * An implementation of HttpAsyncClient that allows direct calling of the
+ * Symfony HttpCache kernel without executing a real HTTP request.
+ * It can only be used if you have a single node setup of Symfony and serves
+ * as kind of a shortcut for easier configuration.
+ * If you use Varnish or have a multiple node Symfony setup, this client is entirely
+ * useless and cannot be used.
+ * It HttPlug libraries to convert between PSR-7 requests and responses. Both are
+ * optional dependencies of this package thus existence of the respective classes
+ * is checked for in the constructor of this client.
+ *
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
+ */
+class KernelClient implements HttpAsyncClient
```

Comment:
>> HttpDispatcher > Dispatcher

> That was my first idea but that doesn't work for BC reasons. Every proxy client expects an instance of HttpDispatcher right now, not of any interface.

this is constructor injection only. so we could change the constructor of HttpProxyClient to say Dispatcher (which works even if some custom extending class says HttpDispatcher because that is a Dispatcher). and the interface has both public methods invalidate & flush, so they are compatible. i think we can sneak this through without any risks for BC?